### ISSUE TITLE
WCAG (AA) - website accessibility 

### ISSUE THREAD

**Issue by tiblu** [citizenos/citizenos-fe-old#71]
https://github.com/rahvaalgatus/rahvaalgatus/issues/28

**Comment by ilmartyrk** [comment]
This is needed for our  UI/UX updates and accessibility. As in the initial description, we could possibly use some starting info from RAA issue and also do some research on latest standarts. This task should probably be research and then listing the things that need to be implemented for WCAG.

**Comment by anettlinno** [comment]
Triage 50. This is important development and need research about the visibility from the WCAG principles. 
@kevincrepin how to design our platform for people with visual impairments. 
What does it mean when generating for HTML @ilmartyrk? 

One option is also to develop frontend tests. 

Sending it to To Do list to investigation. Est. investigation time 1 day.

**Comment by ilmartyrk** [comment]
- [x] Find general standards, list link(s) to documentation/principles (general color guide etc.)
- [x] Find and list guides to generating HTML (alt text etc. that needs to be added to html)
- [x] Check how to create FE  tests and start implementation
- [x] List tools to help with checking compatibility, or libraries that can be used for testing

**Comment by kevincrepin** [comment]
From design
- Higher Contrast settings
- Type size settings
- Plain Text mode (if applicable)
- No animations (if applicable)

**Comment by anettlinno** [comment]
@kevincrepin @ilmartyrk would this be part of the UX/UI redesign project and should we design/develop these together with other things?

**Comment by kevincrepin** [comment]
I would definitely say redesign.

**Comment by anettlinno** [comment]
Triage 70. Est. investigation time 2h.

**Comment by ilmartyrk** [comment]
Standard - WCAG 2.1 (2018) - Standard is described here - https://www.w3.org/TR/WCAG21/
Testing  - https://angular.io/guide/testing
Testing plugin to check WCAG https://www.npmjs.com/package/karma-accessibility-checker
List of tools for WCAG validation - https://www.w3.org/WAI/ER/tools/?q=wcag-21-w3c-web-content-accessibility-guidelines-21&q=en-301-549-european-accessibility-standard (some might not work with angular)

created initial setup for testing in branch called angular-testing. This needs more tweaking and deeper understanding on what and how to evaluate. Current test basically scans through all .html files and as some are just components I get reports on missing html, body tags etc,

**Comment by anettlinno** [comment]
Sending this issue to redesign.

**Comment by devosama1** [comment]
Investigation:

Upon further researching, google provides a list of tools that can be used in angular to test for accessibility such as lighthouse (part of chrome dev tools - automated testing per page), axe and other techniques, For a11y concepts which is a list of concepts that checks for accessibility https://www.a11yproject.com/checklist/ this will ensure that the platform is compliant to latest standards. 

https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd/related?utm_source=deque.com&utm_medium=referral&utm_campaign=axe_hero a good extension to test for accessibility

A really good tutorial for Angular Accessibility: https://www.youtube.com/watch?v=Gm_bD1abFXI

Initial lighthouse testing marked the following for accessibility issues in the platform:
[Lighthouse Report Viewer.pdf](https://github.com/citizenos/citizenos-fe-old/files/11049914/Lighthouse.Report.Viewer.pdf) this is a demo report, what can be expected of it. Might have to do for most of the pages and components manually. Looking for automation alternatives. 

- [ ] Check what kind of screen readers are used by people who require accessibility support. 
- [ ] ARIA is used by screen readers, implementing tags for ARIA will prove very useful. 
- [ ] Check for WCAG checklist
- [ ] Plan a checklist for CitizenOS platform for accessibility (This checklist will prove useful for future components/new blocks)
- [ ] Plan a checklist for CitizenOS website (Though not needed as this would be one time)
- [ ] @angular/cdk/a11y implementation of CDK, which provides further tools in the app for accessibility (Focus trap and live announcer example)
- [ ] Checking for semantic HTML, as most tools use it to understand what the components are. Though material UI provides it by default.
- [ ] Using Karma Accessibility checker for further analysis of the pages.
- [ ] Need of expert review of WCAG of platform, and we can write our own rules based on our needs in Karma Accessibility checker.

**Comment by ilmartyrk** [comment]
Moved to https://github.com/citizenos/citizenos-fe/issues/34
